src/utils/inputfilepath.ts

```diff
@@ -0,0 +1,131 @@
+import * as vscode from 'vscode'
+import * as path from 'path'
+
+import { resolveFile } from './utils'
+
+enum MatchType {
+    Input,
+    Child
+}
+
+interface MatchPath {
+    type: MatchType,
+    path: string,
+    directory: string,
+    matchedString: string,
+    index: number
+}
+
+/**
+ * Return the input regex. This function creates one regex to be used repeatedly
+ * on one document.
+ */
+export function createInputRegExp(): RegExp {
+    return /\\(?:input|InputIfFileExists|include|SweaveInput|subfile|loadglsentries|(?:(?:sub)?(?:import|inputfrom|includefrom)\*?{([^}]*)}))(?:\[[^[\]{}]*\])?{([^}]*)}/g
+}
+
+/**
+ * Return the child regex. This function creates one regex to be used repeatedly
+ * on one Rnw document.
+ */
+export function createChildRegExp(): RegExp {
+    return /<<(?:[^,]*,)*\s*child='([^']*)'\s*(?:,[^,]*)*>>=/g
+}
+
+/**
+ * Return the input and child regexes. This function creates two regexes, each
+ * one is to be used repeatedly on one document.
+ */
+export function createInputChildRegExps(): {input: RegExp, child: RegExp} {
+    return {input: createInputRegExp(), child: createChildRegExp()}
+}
+
+/**
+ * Return the matched input path. If there is no match, return undefined
+ *
+ * @param content the string to match the regex on
+ * @param regexp the input regex created by {@link createInputRegExp()}
+ * @param currentFile is the name of file in which the regex is executed
+ * @param rootFile
+ */
+export function execInputRegExp(content: string, regexp: RegExp, currentFile: string, rootFile: string): {path: string, match: MatchPath} | undefined {
+    const result = regexp.exec(content)
```

Comment:
The `regexp` argument must be the one returned by `createInputRegExp`. The content of `result` is tight to this particular regexp. In particular, if the regexp in `createInputRegExp` is modified, then the parsing should also modified in this function. I think ti would better that the architecture links the two functions together into a class.